File: Data Structure/3.Linked List/Append Linked List.java
/** This is the ListNode class definition

class ListNode {
	int data;
	ListNode next;

	ListNode(int data) {
		this.data = data;
		this.next = null;
	}
}
**/
//https://workat.tech/problem-solving/practice/append-linked-lists/
class Solution {
	ListNode appendLists (ListNode list1, ListNode list2) {
	    // add your logic here
		ListNode currList1 = list1;
		while(currList1.next != null){
			currList1 = currList1.next;
		}
		currList1.next  = list2;
		
		return list1;
	}
}

File: Data Structure/3.Linked List/Append Linked List.py
"""
This is the ListNode class definition

class ListNode:
	def __init__(self, data=0, next=None):
		self.data = data
		self.next = next

"""

# https://workat.tech/problem-solving/practice/append-linked-lists
class Solution:
	def appendLists(self, list1: ListNode, list2: ListNode) -> ListNode:
		# add your logic here
		
		t = list1
		while(t.next!=None):
			t = t.next
			
		t.next  = list2
		
		while(list1 != None):
			print(list1.data,end=' ')
			list1= list1.next
			


